src/main/java/nl/novi/techiteasy/models/Television.java

```diff
@@ -29,33 +30,48 @@ public class Television {
     private Boolean hdr;
     private Boolean bluetooth;
     private Boolean ambiLight;
-    private int originalStock;
-    private int sold;
+    private Integer originalStock;
+    private Integer sold;
     private LocalDate lastSold;
 
-    public Television() {
-    }
-
-    public Television(Long id, String type, String brand, String name, double price, double availableSize, int refreshRate, String screenType, String screenQuality, Boolean smartTv, Boolean wifi, Boolean voiceControl, Boolean hdr, Boolean bluetooth, Boolean ambiLight, int originalStock, int sold, LocalDate lastSold) {
-        this.id = id;
-        this.type = type;
-        this.brand = brand;
-        this.name = name;
-        this.price = price;
-        this.availableSize = availableSize;
-        this.refreshRate = refreshRate;
-        this.screenType = screenType;
-        this.screenQuality = screenQuality;
-        this.smartTv = smartTv;
-        this.wifi = wifi;
-        this.voiceControl = voiceControl;
-        this.hdr = hdr;
-        this.bluetooth = bluetooth;
-        this.ambiLight = ambiLight;
-        this.originalStock = originalStock;
-        this.sold = sold;
-        this.lastSold = lastSold;
-    }
+    @OneToOne(cascade = CascadeType.ALL)
+    private RemoteController remoteController;
+
+    @ManyToOne(fetch = FetchType.EAGER)
```

Comment:
Het is niet handig om FetchType.Eager als default te gebruiken. Dit is alleen voor relaties die onlosmakelijk met elkaar zijn verbonden. Dat is hier niet het geval. Je werkt hier met een bidirectionele relatie (Television weet van CiModule en CiModule weet van Television), wat in combinatie met je FetchType.Eager wel eens de recursie zou kunnen veroorzaken die je noemde.